core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -51,22 +57,35 @@ object MultiCloudWorkspaceService {
       dataSource,
       workbenchMetricBaseName
     )
+
+  def getCloudPlatform(p: RawlsBillingProject): WorkspaceCloudPlatform =
+    AzureBillingProject.unapply(p).as(Azure).getOrElse(Gcp)
+
+  object AzureBillingProject {
+    def unapply(p: RawlsBillingProject): Option[(RawlsBillingProjectName, UUID)] =
+      for {
+        profileId <- p.billingProfileId
+        uuid <- Try(UUID.fromString(profileId)).toOption
```

Comment:
I'm not sure if it's better to throw `InternalServerError` if the profile id is malformed - I'm guessing this is something we create ourselves and the user has no control over the billing project id?

Reply:
As discussed, if the profileId exists, you should look up the billing profile record to get the cloud platform. Otherwise assume GCP (example: https://github.com/broadinstitute/rawls/blob/20575d547dbab706c5b712c5548203acbc8d34a9/core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala#L315)

And yes, I think an InternalServerError would be appropriate if profileId is malformed.